src/components/data/WdsTroubleshooter.js

```diff
@@ -0,0 +1,202 @@
+import _ from 'lodash/fp'
+import { Fragment, useState } from 'react'
+import { div, h, table, td, tr } from 'react-hyperscript-helpers'
+import { ClipboardButton } from 'src/components/ClipboardButton'
+import { ButtonPrimary } from 'src/components/common'
+import { icon } from 'src/components/icons'
+import Modal from 'src/components/Modal'
+import { Ajax } from 'src/libs/ajax'
+import { resolveWdsApp } from 'src/libs/ajax/data-table-providers/WdsDataTableProvider'
+import colors from 'src/libs/colors'
+import { useCancellation, useOnMount } from 'src/libs/react-utils'
+import * as Style from 'src/libs/style'
+import * as Utils from 'src/libs/utils'
+
+
+export const WdsTroubleshooter = ({ onDismiss, workspaceId, mrgId }) => {
+  const [leoOk, setLeoOk] = useState(null)
+  const [wdsResponsive, setWdsResponsive] = useState(null)
+  const [version, setVersion] = useState(null)
+  const [wdsStatus, setWdsStatus] = useState(null)
+  const [wdsDbStatus, setWdsDbStatus] = useState(null)
+  const [wdsPingStatus, setWdsPingStatus] = useState(null)
+  const [appFound, setAppFound] = useState(null)
+  const [appRunning, setAppRunning] = useState(null)
+  const [proxyUrl, setProxyUrl] = useState(null)
+  const [defaultInstanceExists, setDefaultInstanceExists] = useState(null)
+
+  const signal = useCancellation()
+
+  useOnMount(() => {
+    Ajax(signal).Apps.getV2AppInfo(workspaceId).then(res => {
```

Comment:
We are already calling this once upon the constructor in `WDSDataTableProvider` -- can you pass it as a prop. instead to this component so we don't need two identical API calls?

Reply:
Maybe... although it might depend on when/how often each call is made and in what order - if there have been any changes to what apps exist or if they've changed from PROVISIONING to RUNNING I'd want to get new info from Leo.  If we're still not polling, will `WDSDataTableProvider` only be constructed once unless we reload the page?

Reply:
I'm wondering if it's a feature to have the troubleshooter trigger a call on-demand when the user opens it. Thoughts?

Reply:
I like having the troubleshooter re-do all the API calls. That way, if we see the troubleshooter return all green even though the underlying page has errors, we can recommend an appropriate action: reload your browser

Reply:
Resolving my comment here -- seems what @davidangb has in mind here is a good approach